packages/http-client-csharp/emitter/src/lib/operation-converter.ts

```diff
@@ -373,18 +363,28 @@ function getParameterKind(
   hasGlobalApiVersion: boolean
 ): InputOperationParameterKind {
   if (p.kind === "body") {
-    if (type.kind === "model" && (type.usage & UsageFlags.Spread) !== 0) {
+    /** TODO: remove this and use the spread metadata of parameter when https://github.com/Azure/typespec-azure/issues/1513 is resolved */
+    if (type.kind === "model" && p.type !== p.correspondingMethodParams[0]?.type) {
       return InputOperationParameterKind.Spread;
     }
     return InputOperationParameterKind.Method;
   }
+
+  /** remove this, use p.onClient directly when https://github.com/Azure/typespec-azure/issues/1532 is resolved */
+  const paramOnClient =
+    p.correspondingMethodParams &&
+    p.correspondingMethodParams.length > 0 &&
+    p.correspondingMethodParams[0].onClient;
+
   return type.kind === "constant"
     ? InputOperationParameterKind.Constant
     : p.isApiVersionParam
       ? hasGlobalApiVersion
```

Comment:
maybe we don't need to check global api version as well, see comments: https://github.com/Azure/typespec-azure/issues/1489#issuecomment-2341012488

Reply:
I think check `hasGlobalApiVersion` is to bypass the issue https://github.com/Azure/typespec-azure/issues/1391. We will remove all those workaround when TCGC resolve all those.